src/builder/modules/Generator_Builder.res

```diff
@@ -437,6 +437,48 @@ let fillBackgroundColor = (model: Model.t, fillStyle: string) => {
   }
 }
 
+let drawLine = (
+  model: Model.t,
+  (dx, dy, dw, dh): rectangle,
+  ~color: string,
+  ~width: float,
+  ~pattern: array<int>,
+  ~offset: int,
+) => {
+  let x = Belt.Int.toFloat(dx)
+  let y = Belt.Int.toFloat(dy)
+  let w = Belt.Int.toFloat(dw)
+  let h = Belt.Int.toFloat(dh)
+
+  let angle = w === 0.0 ? Js.Math._PI /. 2.0 : Js.Math.atan2(~y=h, ~x=w, ())
+  let ow = Js.Math.sin(angle) *. 0.5
+  let oh = Js.Math.cos(angle) *. 0.5
+
+  switch model.currentPage {
+  | None => model
+  | Some(currentPage) => {
+      let currentPage = findPage(model, currentPage.id)
+      switch currentPage {
+      | None => model
+      | Some(currentPage) => {
+          let context = currentPage.canvasWithContext.context
+          context->Context2d.setImageSmoothingEnabled(false)
```

Comment:
As you mentioned, I believe this only affects images, not lines, so we can remove it?